Fixes #15046 - Add bmc_nic_setup snippet

bagasse opened the pull request:
Template part of http://projects.theforeman.org/issues/15046.

Commit:
Fixes #15046 - Add bmc_nic_setup snippet

mmoll commented:
This is looking very useful, but I don't have any hardware to test this, so only two comments from my side:
- Please make this snippet more OS agnostic. Have a look onto the puppet_setup snippet as an example. I'd like to see that working also on Debian/Ubuntu, idealy also on FreeBSD. Especially regarding the later (no GNU grep or awk) I'd recommend e.g. to use cut instead of awk and `tail -n 1` instead of `tail -1`
- I _think_ there's also hardware on the market that doesn't use 1 as its default channel (@lzap I think you came across some IBM h/w where that was the case), so it might make sense to extract that into a parameter.

bagasse commented:
@mmoll, thank you for your comment. I've only dell hardware so i didn't test it on other hardware. I've modified the snippet to:
- Manage channel IPMI channel ID and use 1 by default
- Use package manager from os type family (maybe i have to manage dnf for recent fedora releases ?).
- Use portable options of grep, and replaced awk with cut

Modifications tested on CentOS7 at this time. No hardware test server free at this time, it will be difficult to me to test it on some debian or freebsd versions.

lzap commented:
I also don't have IPMI enabled hardware myself, but since this is a snippet, I think we can merge it.

mmoll commented on snippets/bmc_nic_setup.erb:
```diff
@@ -0,0 +1,97 @@
+<%#
+kind: snippet
+name: bmc_nic_setup
+oses:
```
snippets don't have OS associations, so this should get deleted

mmoll commented on snippets/bmc_nic_setup.erb:
```diff
@@ -0,0 +1,97 @@
+# It was tested on some Dell hardware with CentOS 6 - CentOS 7 and should be compatible with respective
+# upstream RHEL versions.
+#
+# What this snippet do:
```
does

mmoll commented on snippets/bmc_nic_setup.erb:
```diff
@@ -0,0 +1,97 @@
+#
+# * Enable the user created/updated
+#
+# What this snippet don't do:
```
doesn't

mmoll commented on snippets/bmc_nic_setup.erb:
```diff
@@ -0,0 +1,97 @@
+# Configure BMC interface
+<% case os_family %>
+<% when 'Redhat' then %>
+yum  -t -y install ipmitool
```
one space too much here

mmoll commented on snippets/bmc_nic_setup.erb:
```diff
@@ -0,0 +1,97 @@
+os_family = @host.operatingsystem.family
+os_major  = @host.operatingsystem.major.to_i
+os_name   = @host.operatingsystem.name
+
```
major and name are not used, AFAICS?

mmoll commented:
There's no dnf usage in any of the templates, yet, so that's OK

bagasse commented:
Done.

mmoll commented:
merged, merci @bagasse!

bagasse commented:
@mmoll, @lzap, thank you for comments/reviews
